### PR TITLE
Fetch the llvm-spirv component automatically if no llvm-spirv specified.

### PR DESCRIPTION
**Pull request by chengjunlu**
Add the public llvm-spirv component to the Triton XPU backend. 
Support to build the XPU backend friendly.

### CODE REVIEW

File: lib/Target/SPIRV/CMakeLists.txt [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# SPIRV-LLVM Translator is required.
```
Comment (whitneywhtsang) [inline lib/Target/SPIRV/CMakeLists.txt:1]: nit: consistent name
```suggestion
# SPIRV-LLVM-Translator is required.
```

**Review by etiotto — APPROVED** [review]
LGTM

File: lib/Target/SPIRV/spirv-llvm-translator.conf [hunk 2]
```diff
@@ -0,0 +1 @@
+08885516deabe3b51c8bc801fd8997b794e9d698
```
Comment (whitneywhtsang) [inline lib/Target/SPIRV/spirv-llvm-translator.conf:1]: Any reasons you picked this commit? 0166a0fb86dc6c0e8903436bbc3a89bc3273ebc0 is later, and it works.

File: cmake/FindSPIRVToLLVMTranslator.cmake [hunk 3]
```diff
@@ -0,0 +1,32 @@
+# - Try to find SPIRV-LLVM-Translator
```
Comment (whitneywhtsang) [inline cmake/FindSPIRVToLLVMTranslator.cmake:1]: nit
```suggestion
# Try to find SPIRV-LLVM-Translator
```

File: lib/Target/SPIRV/spirv-llvm-translator.conf (hunk 2 above)
Comment (etiotto) [reply lib/Target/SPIRV/spirv-llvm-translator.conf:1]: We will need to pick a particular commit  now and periodically move the commit toward the latest. Whatever commit is latest now should work.

Comment (whitneywhtsang) [reply lib/Target/SPIRV/spirv-llvm-translator.conf:1]: I am in the process of finding the latest commit that works.

Comment (whitneywhtsang) [reply lib/Target/SPIRV/spirv-llvm-translator.conf:1]: adf1f524ac82f3206536ef6fc6fbc289a900e5d6 should be the last commit that still works.

**Comment by whitneywhtsang** [comment]
https://github.com/intel/intel-xpu-backend-for-triton/blob/llvm-target/scripts/compile-triton.sh can be modified to no longer clone the SPIRV-LLVM translator after this change. Created https://github.com/intel/intel-xpu-backend-for-triton/pull/189.

File: lib/Target/SPIRV/spirv-llvm-translator.conf (hunk 2 above)
Comment (chengjunlu) [reply lib/Target/SPIRV/spirv-llvm-translator.conf:1]: Update the revision to adf1f524ac82f3206536ef6fc6fbc289a900e5d6

File: lib/Target/SPIRV/CMakeLists.txt [hunk 4]
```diff
@@ -6,4 +9,9 @@ add_mlir_translation_library(TritonSPIRV
 
         LINK_LIBS PUBLIC
         TritonLLVMIR
+        # spirv tools
+        LLVMSPIRVLib
         )
+
+# Add SPIRVLLVMTranslator include dir
```
[the remark this answers is not in the document]
Comment (chengjunlu) [reply lib/Target/SPIRV/CMakeLists.txt]: Code changed correspondingly.

File: lib/Target/SPIRV/CMakeLists.txt (hunk 1 above)
Comment (chengjunlu) [reply lib/Target/SPIRV/CMakeLists.txt]: Code changed correspondingly.

File: cmake/FindSPIRVToLLVMTranslator.cmake (hunk 3 above)
Comment (chengjunlu) [reply cmake/FindSPIRVToLLVMTranslator.cmake]: Code changed correspondingly.

**Review by etiotto — APPROVED** [review]

**Review by whitneywhtsang — APPROVED** [review]

**Pull request merged**